IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -387,42 +440,107 @@ - (BOOL)removeAccountsWithKey:(MSIDCacheKey *)key
 #pragma mark - Credentials
 
 // Write a credential to the macOS keychain cache.
-- (BOOL)saveToken:(__unused MSIDCredentialCacheItem *)credential
-              key:(__unused MSIDCacheKey *)key
-       serializer:(__unused id<MSIDCacheItemSerializing>)serializer
-          context:(__unused id<MSIDRequestContext>)context
-            error:(__unused NSError **)error
+- (BOOL)saveToken:(MSIDCredentialCacheItem *)credential
+              key:(MSIDCacheKey *)key
+       serializer:(id<MSIDCacheItemSerializing>)serializer
+          context:(id<MSIDRequestContext>)context
+            error:(NSError **)error
 {
     assert(credential);
     assert(serializer);
-    [self updateLastModifiedForCredential:credential context:context];
-    NSData *itemData = [serializer serializeCredentialCacheItem:credential];
+    
+    if ([key isKindOfClass:[MSIDDefaultCredentialCacheKey class]])
```

Comment:
nit: log if the key is not of correct class to help us catch the issues

Reply:
Updated.